File: gists/001ce73aec7bca26ec5d95654232157f/go/main.go
package main

import (
	"fmt"
	"time"
)

func calc(n uint64) float64 {
	piOver4 := 0.0
	for i := uint64(0); i < n; i++ {
		term := 1.0 / (2.0*float64(i) + 1.0)
		if i%2 == 0 {
			piOver4 += term
		} else {
			piOver4 -= term
		}
	}
	return piOver4 * 4.0
}

func main() {
	sum := 0.0
	for i := 0; i < 20; i++ {
		start := time.Now()

		for i := 0; i < 1000; i++ {
			sum += calc(1000000)
		}

		duration := time.Since(start)
		fmt.Println(duration)
	}
	fmt.Println(sum)
}

File: gists/001ce73aec7bca26ec5d95654232157f/rs/src/main.rs
use std::time::Instant;
use std::hint::black_box;

pub fn calc(n: u64) -> f64 {
	let mut pi_over_4 = 0.0;
	for i in 0..n {
		let term = 1.0 / (2.0 * i as f64 + 1.0);
		if i % 2 == 0 {
			pi_over_4 += term;
		} else {
			pi_over_4 -= term;
		}
	}
	pi_over_4 * 4.0
}

fn main() {
	for _ in 0..20 {
		let start = Instant::now();

		for _ in 0..1000 {
			black_box(calc(1000000));
		}

		let duration = start.elapsed();
		println!("{:?}", duration);
	}
}
